pkgs/tools/nix/nix-output-monitor/default.nix

```
@@ -25,6 +26,10 @@ mkDerivation {
     ansi-terminal async attoparsec base containers directory HUnit mtl
     nix-derivation process relude stm text time unix
   ];
+  postInstall = ''
+    echo -e "#!/bin/sh\n${expect}/bin/unbuffer nix-build \"\$@\" 2>&1 | exec $out/bin/nom" > $out/bin/nom-build
```

```suggestion
    echo -e "#!/bin/sh
${expect}/bin/unbuffer nix-build \"\$@\" |& $out/bin/nom
" > $out/bin/nom-build
```

Why do we want exec here?

---

idk, I don‘t know of a disadvantage and I don‘t see that we need the running bash process after nom has been started. Also maybe exit code propagation?

---

`exec` is really just to get rid of the bash process, there’s no reason to keep it around. It also will keep the `ps` output cleaner.

---

This would probably also be a bit cleaner if written using a here-doc. At a bare minimum, I don’t see why this is using the `-e` flag to `echo`. And the shebang should probably be `#!${runtimeShell}`.

---

Heck, you could probably make this a `writeShellScript` instead, using the `outPath` attribute of the resulting derivation, like

```nix
let
  nom-build = writeShellScript "nom-build" ''
    ${expect}/bin/unbuffer nix-build "$@" |& ${self}/bin/nom
  '';
  self = mkDerivation rec {
    # …
    postInstall = ''
      install -Dm755 ${nom-build} $out/bin/nom-build
    '';
  };
in self
```

I haven’t tried this directly myself, but I _think_ the `outPath` is available without forcing anything beyond `name` and fixed-hash attrs.

---

If thought about that, but was un-optimistic. I’ll try.

---

No, that doesn‘t make sense. If the hash of nom changed the hash of nom-build needs to change and vice versa. I‘ll try anyways just to make sure, but I would be thoroughly surprised.

---

Thanks for the feedback, I have rewritten the function with your hints. Looks much better now.